examples/python-cicd-with-cloudbuilder/python-cloud-builder/cloudbuild.yaml

```diff
@@ -0,0 +1,8 @@
+
```

Comment:
Again, another YAML file. Maybe some comments in here (and in the dockerfile) about how everything fits together locally and/or when run in could build?

Reply:
Also, it seems like this is a much less rich yaml than the example one. Why? Again, I am maybe misunderstanding here.

Reply:
I can see why these are crazy confusing because this was really confusing for me too. 

Step 3 of the README shows the reader how to build a custom cloud builder image. The files in python-cloud-builder/ create that custom image, using cloud builder...so it's a bit of cloud builder inception.

Once the custom cloud builder in python-cloud-builder/ is built, the cloudbuild.yaml in the root calls the custom cloud builder to run the CI steps.

It's really a little awkward because google doesn't distribute a python cloud build image (and i don't know why that is, other than using virtualenvs to carry work forward is a little opinionated)

Reply:
There's a comment above about this too (sorry). I think you need to call this out specifically in the README file, that you build a base python cloud builder container with the YAML in the directory (and maybe walk briefly through that yaml like you do with the main one) and then you use the root yaml + the built container to run the CICD pipeline. Maybe its worth giving one of the yaml files a different name to make it clearer (or giving both unique names?), though that may not be possible.

Thanks for rolling with this, I know my asks are extensive, and the naming schemes and interface of cloud build make things a bit difficult. You're doing important work here.

Reply:
I got way more explicit on this under ## 3.   Hopefully it's clearer now.